tensornetwork/visualization/graphviz.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Implementation of TensorNetwork Graphviz visualization."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import graphviz
+from typing import Optional, Text
+from tensornetwork import network
+
+
+def to_graphviz(
+    net: network.TensorNetwork, 
+    graph: Optional[graphviz.Graph] = None,
+    include_all_names: bool=False,
+    engine: Text = "neato"
+  ) -> graphviz.Graph:
+  """Create a graphviz Graph that is isomorphic to the given TensorNetwork.
+
+  Args:
+    net: A `TensorNetwork`.
+    graph: An optional `graphviz.Graph` object to write to. Use this only
+      if you wish to set custom attributes for the graph.
+    include_all_names: Whether to include all of the names in the graph.
+      If False, all names starting with '__' (which are almost always just
+      the default generated names) will be dropped to reduce clutter.
+
+  Returns:
+    The `graphviz.Graph` object.
+  """
+  if graph is None:
+    graph = graphviz.Graph('G', engine=engine)
+  for node in net.nodes_set:
+    label = node.name if node.name[:2] != "__" or include_all_names else ""
+    graph.node(str(node.signature), label=label)
+  seen_edges = set()
+  for node in net.nodes_set:
+    for i, edge in enumerate(node.edges):
+      if edge not in seen_edges:
+        seen_edges.add(edge)
+        if edge.name[:2] != "__" or include_all_names:
```

Comment:
nit: Maybe `not edge.name.startswith("__")`?

Reply:
Done